common/core/src/authentication_provider.ts

```diff
@@ -23,5 +23,5 @@ export enum AuthenticationType {
  */
 export interface AuthenticationProvider {
   type: AuthenticationType;
-  getDeviceCredentials(callback: (err: Error, credentials?: TransportConfig) => void): void;
+  getDeviceCredentials(callback?: (err?: Error, credentials?: TransportConfig) => void): void;
```

Comment:
I don't get it... the callback should not be optional. the typescript error described in #403 seem to indicate a problem between the DoubleValueCallback type and the current callback type, not with the fact that it's optional. 
or is that a fix for something else?

Reply:
Done.

---
In reply to: [228321598](https://github.com/Azure/azure-iot-sdk-node/pull/405#discussion_r228321598) [](ancestors = 228321598)

Reply:
This permits us to use the whole promise thing.

It is a reasonable question as to whether these providers should be promisified, but since that already happened, I think this just completes the job.

---
In reply to: [228325158](https://github.com/Azure/azure-iot-sdk-node/pull/405#discussion_r228325158) [](ancestors = 228325158)